libraries/core/src/test/java/com/paritytrading/philadelphia/FIXInitiatorTest.java

```diff
@@ -44,31 +45,29 @@ class FIXInitiatorTest {
     private FIXMessages  initiatorMessages;
     private TestMessages acceptorMessages;
 
-    private FIXConnectionStatus initiatorStatus;
+    private FIXConnectionStatus<INITIATOR_CHAN> initiatorStatus;
 
-    private FIXConnection  initiator;
+    private FIXConnection<INITIATOR_CHAN> initiator;
     private TestConnection acceptor;
 
-    @BeforeEach
-    void setUp() throws IOException {
-        ServerSocketChannel acceptorServerChannel = ServerSocketChannel.open();
-        acceptorServerChannel.bind(null);
-
-        SocketChannel initiatorChannel = SocketChannel.open(acceptorServerChannel.getLocalAddress());
-        initiatorChannel.configureBlocking(false);
-
-        SocketChannel acceptorChannel = acceptorServerChannel.accept();
-        acceptorChannel.configureBlocking(false);
+    protected final class Channels {
+        public INITIATOR_CHAN initiator;
+        public ByteChannel acceptor;
+    }
 
-        acceptorServerChannel.close();
+    @NotNull
+    protected abstract Channels createChannels() throws IOException;
```

Comment:
This method could be in the package-private scope like the other non-private members in the class.

Reply:
Generally `protected` communicates that you intend this to be called from subclasses regardless of where they are located. This is our intent with this method and the class above - they are part of our interface to subclasses.

That said, each project has its peculiarities, and I see no harm in this.